File: robocode-tankroyale-bot-api-java/src/main/java/dev/robocode/tankroyale/botapi/events/SkippedTurnEvent.java
package dev.robocode.tankroyale.botapi.events;

/**
 * Event occurring when the bot has skipped a turn, meaning that no intent has reached the server
 * for a specific turn.
 */
public final class SkippedTurnEvent extends BotEvent {

  /**
   * Initializes a new instance of the SkippedTurnEvent class.
   *
   * @param turnNumber is the turn number which was skipped.
   */
  public SkippedTurnEvent(int turnNumber) {
    super(turnNumber);
  }
}

File: robocode-tankroyale-bot-api-dotnet/src/events/SkippedTurnEvent.cs
using Newtonsoft.Json;

namespace Robocode.TankRoyale.BotApi.Events
{
  /// <summary>
  /// Event occurring when the bot has skipped a turn, meaning that no intent has reached the server
  /// for a specific turn.
  /// </summary>
  public sealed class SkippedTurnEvent : BotEvent
  {
    /// <summary>
    /// Initializes a new instance of the SkippedTurnEvent class.
    /// </summary>
    /// <param name="turnNumber">Turn number.</param>
    [JsonConstructor]
    public SkippedTurnEvent(int turnNumber) : base(turnNumber) { }
  }
}